packages/wordpress-block-editor-config/src/settings/__experimental-fetch-link-suggestions.js

```diff
@@ -0,0 +1,152 @@
+/**
+ * WordPress dependencies
+ */
+// @ts-ignore TODO
+import apiFetch from '@wordpress/api-fetch';
+import { addQueryArgs } from '@wordpress/url';
+import { decodeEntities } from '@wordpress/html-entities';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Filters the search by type
+ *
+ * @typedef { 'post' | 'term' | 'post-format' } WPLinkSearchType
+ */
+
+/**
+ * @typedef WPLinkSearchOptions
+ *
+ * @property {?boolean}              isInitialSuggestions Displays initial search suggestions, when true.
+ * @property {?WPLinkSearchType}     type                 Filters by search type.
+ * @property {?string}               subtype              Slug of the post-type or taxonomy.
+ * @property {?number}               page                 Which page of results to return.
+ * @property {?number}               perPageArg              Search results per page.
+ */
+
+/**
+ * @typedef WPLinkSearchResult
+ *
+ * @property {?number} id     Post or term id.
+ * @property {?string} url    Link url.
+ * @property {?string} title  Title of the link.
+ * @property {?string} type   The taxonomy or post type slug or type URL.
+ */
+
+/**
+ * Fetches link suggestions from the API.
+ *
+ * @async
+ * @typedef {(searchText:string,searchOptions:WPLinkSearchOptions)=>Promise<WPLinkSearchResult[]>} WPFetchLinkSuggestions
+ */
+
+/**
+ * Returns a function that when invoked, fetches link suggestions from the API.
+ *
+ * @param {Object}   [editorSettings]
+ * @param {?boolean} [editorSettings.disablePostFormats]
```

Comment:
Note that `?boolean` is redundant if the property is optional unless `disablePostFormats: null` is a valid option to explicitly declare.

Reply:
Thanks, I didn't realize the brackets were for optional parameters in JSDoc. I've updated this in 9cfa345